samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/MultiLingualBot.java

```diff
@@ -0,0 +1,186 @@
+    }
+
+    /**
+     * This method is executed when the turnContext receives a message activity.
+     * @param turnContext The context object for this turn.
+     * @return A task that represents the work queued to execute.
+     */
+    @Override
+    protected CompletableFuture<Void> onMessageActivity(TurnContext turnContext) {
+        if (MultiLingualBot.isLanguageChangeRequested(turnContext.getActivity().getText())) {
+            String currentLang = turnContext.getActivity().getText().toLowerCase();
+            String lang = currentLang.equals(ENGLISH_ENGLISH) || currentLang.equals(SPANISH_ENGLISH)
+                ? ENGLISH_ENGLISH : ENGLISH_SPANISH;
+
+            // If the user requested a language change through the suggested actions with values "es" or "en",
+            // simply change the user's language preference in the user state.
+            // The translation middleware will catch this setting and translate both ways to the user's
+            // selected language.
+            // If Spanish was selected by the user, the reply below will actually be shown in spanish to the user.
+            return languagePreference.set(turnContext, lang)
+                .thenCompose(task -> {
+                    Activity reply = MessageFactory.text(String.format("Your current language code is: %s", lang));
+                    return turnContext.sendActivity(reply);
+                })
+                // Save the user profile updates into the user state.
+                .thenCompose(task -> userState.saveChanges(turnContext, false));
+        } else {
+            // Show the user the possible options for language. If the user chooses a different language
+            // than the default, then the translation middleware will pick it up from the user state and
+            // translate messages both ways, i.e. user to bot and bot to user.
+            Activity reply = MessageFactory.text("Choose your language:");
+            CardAction esAction = new CardAction() {
+                {
+                    setTitle("Español");
+                    setType(ActionTypes.POST_BACK);
+                    setValue(ENGLISH_SPANISH);
+                }
+            };
+            CardAction enAction = new CardAction() {
+                {
+                    setTitle("English");
+                    setType(ActionTypes.POST_BACK);
+                    setValue(ENGLISH_ENGLISH);
+                }
+            };
+            List<CardAction> actions = new ArrayList<>(Arrays.asList(esAction, enAction));
+            SuggestedActions suggestedActions = new SuggestedActions() {
+                {
+                    setActions(actions);
+                }
+            };
+            reply.setSuggestedActions(suggestedActions);
+            return turnContext.sendActivity(reply).thenApply(resourceResponse -> null);
+        }
+    }
+
+    private static CompletableFuture<Void> sendWelcomeMessage(TurnContext turnContext) {
```

Comment:
sendWelcomeMessage will only send to the first non-target member.  This probably works fine when there is only one.

For multiples, need to build a list of CompletableFuture.

Something like:

```    
private static CompletableFuture<Void> sendWelcomeMessage(TurnContext turnContext) {
        return turnContext.getActivity().getMembersAdded().stream()
            .filter(
                member -> !StringUtils
                    .equals(member.getId(), turnContext.getActivity().getRecipient().getId())
            )
            .map(channel -> {
                Attachment welcomeCard = MultiLingualBot.createAdaptiveCardAttachment();
                Activity response = MessageFactory.attachment(welcomeCard);
                return turnContext.sendActivity(response)
                    .thenCompose(task -> turnContext.sendActivity(MessageFactory.text(WELCOME_TEXT)));
            })
            .collect(CompletableFutures.toFutureList())
            .thenApply(resourceResponses -> null);
}
```

Reply:
Thanks @tracyboehrer, we will be reviewing the requested change and we will update the PR 😊.

Reply:
@tracyboehrer - we already applied the requested changes in this [commit](https://github.com/microsoft/botbuilder-java/pull/898/commits/88293caf4e959199380177b52f9e6b7c3be499c5).

We are currently validating the functionality, as soon as we finish with the validation we will update the PR to be out of draft.